pkg/controller/status/status.go

```diff
@@ -312,8 +337,7 @@ func (m *statusManager) IsProgressing() bool {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	if m.progressing == nil || m.failing == nil {
```

Comment:
`if m.progressing == nil || m.failing == nil || !m.readyToMonitor`

The way the code is right now, it probably doesn't matter, since syncState will always complete. But the function contract doesn't guarantee that will always be the case (or has always been the case).